antarest/study/storage/variantstudy/variant_study_service.py

```diff
@@ -0,0 +1,277 @@
+        self, study_id: str, command_id: str, new_parent_id: str
+    ) -> None:
+        """
+        Move command place in the list of command
+        Args:
+            study_id: study id
+            command_id: command_id
+            new_parent_id: new parent id of the command
+        Returns: None
+        """
+        raise NotImplementedError()
+
+    def remove_command(self, study_id: str, command_id: str) -> None:
+        """
+        Remove command
+        Args:
+            study_id: study id
+            command_id: command_id
+        Returns: None
+        """
+        raise NotImplementedError()
+
+    def update_command(
+        self, study_id: str, command_id: str, command: CommandDTO
+    ) -> None:
+        """
+        Update a command
+        Args:
+            study_id: study id
+            command_id: command id
+            command: new command
+        Returns: None
+        """
+        raise NotImplementedError()
+
+    def get_study(self, study_id: str) -> FileStudy:
+        """
+        Get study
+        Args:
+            study_id: study id
+        Returns: Study
+        """
+        raise NotImplementedError()
+
+    def import_output(self, study: Study, output: Union[bytes, Path]) -> None:
+        """
+        Import an output
+        Args:
+            study: the study
+            output: Path of the output or raw data
+        Returns: None
+        """
+        raise NotImplementedError()
+
+    def create_variant_study(
+        self, uuid: str, name: str, params: RequestParameters
+    ) -> Optional[str]:
+        """
+        Create empty study
+        Args:
+            uuid: study name to set
+            name: name of study
+            params: request parameters
+
+        Returns: new study uuid
+
+        """
+        study = self.repository.get(uuid)
+
+        if study is None:
+            raise StudyNotFoundError(uuid)
+
+        assert_permission(params.user, study, StudyPermissionType.READ)
+        new_id = str(uuid4())
+        study_path = str(self.get_default_workspace_path() / new_id)
+        variant_study = VariantStudy(
+            id=new_id,
+            name=name,
+            parent_id=uuid,
+            workspace=study_path,
+            path=study_path,
+            public_mode=study.public_mode,
+            created_at=datetime.now(),
+            updated_at=datetime.now(),
+            version=study.version,
+            groups=study.groups,  # Create inherit_group boolean
+            owner=params.user,
+            snapshot=None,
+        )
+
+        self.repository.save(variant_study)
+        self.event_bus.push(
+            Event(EventType.STUDY_CREATED, variant_study.to_json_summary())
+        )
+
+        logger.info(
+            "variant study %s created by user %s",
+            variant_study.id,
+            params.get_user_id(),
+        )
+        return str(variant_study.id)
+
+    def get_workspace_path(self, workspace: str) -> Path:
```

Comment:
this is duplicated from raw_study_service, should create a static utility function that takes the config in parameters